mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2512,8 +2512,7 @@ pub async fn check_balance_for_taker_swap(
     let params = match prepared_params {
         Some(params) => params,
         None => {
-            // Use None as taker_pubkey is okay because we just need to calculate max swap amount
-            let dex_fee = DexFee::new_from_taker_coin(my_coin, other_coin.ticker(), &volume, None);
+            let dex_fee = DexFee::new_from_taker_coin(my_coin, other_coin.ticker(), &volume); // taker_pubkey is not known yet so we get max dexfee to estimate max swap amount
```

Comment:
I don't understand how taker pubkey is not know yet when this function is called by the taker only and the taker should know their own pubkey.

Reply:
bad comment, will fix it.
It was meant taker_pub may be optional (for dexfee calculation we can use taker_pub for non-privacy coins only)